### PR TITLE
As an Admin, I want to modify taxons of a product

### PR DESCRIPTION
**Pull request by everwhatever**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13 |
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no  |
| Deprecations?   | no |
| Related tickets | [jira]( https://sylius.atlassian.net/browse/SYL-2215)           |
| License         | MIT                                                          |

### CODE REVIEW

File: features/promotion/applying_catalog_promotions/reapplying_catalog_promotions_on_variants_once_products_taxon_changes.feature [hunk 1]
```diff
@@ -21,7 +21,7 @@ Feature: Reapplying catalog promotions on variants once the product’s taxon ch
         When I change that the "T-Shirt" product does not belong to the "Clothes" taxon
         Then the visitor should see that the "PHP T-Shirt" variant is not discounted
 
-    @ui
+    @ui @api
     Scenario: Adding a taxon to a product
-        When I change that the "Mug" product belongs to the "Dishes" taxon
+        When I add "Dishes" taxon to the "Mug" product
```
Comment (jakubtobiasz) [inline features/promotion/applying_catalog_promotions/reapplying_catalog_promotions_on_variants_once_products_taxon_changes.feature]: **Not a hard opinion**
```suggestion
        When I add the "Mug" product to the "Dishes" taxon
```
or
```suggestion
        When I assign the "Dishes" taxon to the "Mug" product
```

Add a taxon to a product sound weird to me.

**Review by jakubtobiasz — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Resources.php [hunk 2]
```diff
@@ -53,6 +53,8 @@ final class Resources
 
     public const PRODUCT_OPTIONS = 'product-options';
 
+    public const PRODUCT_TAXONS = 'product-taxons';
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Resources.php]: You can move it to be in alphabetical order

File: src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php [hunk 3]
```diff
@@ -655,6 +655,17 @@ public function thisProductMainTaxonShouldBe(ProductInterface $product, $taxonNa
         Assert::true($currentPage->isMainTaxonChosen($taxonName));
     }
 
+    /**
+     * @Then /^the (product "[^"]+") should have the "([^"]+)" taxon$/
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php]:
```suggestion
     * @Then the product :product should have the :taxonName taxon
```

File: src/Sylius/Behat/Resources/config/suites/api/promotion/applying_catalog_promotions.yml [hunk 4]
```diff
@@ -15,6 +15,7 @@ default:
                 - sylius.behat.context.setup.taxonomy
                 - Sylius\Behat\Context\Setup\CatalogPromotionContext
                 - Sylius\Calendar\Tests\Behat\Context\Setup\CalendarContext
+                - Sylius\Behat\Context\Api\Admin\ManagingProductTaxonsContext
```
Comment (GSadee) [inline src/Sylius/Behat/Resources/config/suites/api/promotion/applying_catalog_promotions.yml]: Move it to be after other API contexts (line 33)

File: src/Sylius/Behat/Resources/config/suites/ui/product/managing_products.yml [hunk 5]
```diff
@@ -42,6 +42,7 @@ default:
                 - sylius.behat.context.setup.shipping_category
                 - sylius.behat.context.setup.taxonomy
                 - sylius.behat.context.setup.zone
+                - Sylius\Behat\Context\Ui\Admin\ManagingProductTaxonsContext
```
Comment (GSadee) [inline src/Sylius/Behat/Resources/config/suites/ui/product/managing_products.yml]: Move it to be after other UI contexts

File: src/Sylius/Bundle/ApiBundle/DataPersister/ProductTaxonDataPersister.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+use Sylius\Component\Core\Model\ProductTaxonInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+/** @experimental */
+final class ProductTaxonDataPersister implements ContextAwareDataPersisterInterface
+{
+    public function __construct(
+        private ContextAwareDataPersisterInterface $decoratedDataPersister,
+        private MessageBusInterface $eventBus
+    ){
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/DataPersister/ProductTaxonDataPersister.php]:
```suggestion
        private MessageBusInterface $eventBus,
    ) {
```

**Review by GSadee — COMMENTED** [review]
If I see correctly, after the last changes, one of the steps is not defined: https://github.com/Sylius/Sylius/actions/runs/3938703688/jobs/6737710320#step:25:406

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, @everwhatever! :1st_place_medal:

**Pull request merged**